napi-inl.h

```diff
@@ -2193,47 +2175,38 @@ inline void AsyncWorker::Cancel() {
   if (status != napi_ok) throw Error::New(_env);
 }
 
-inline void AsyncWorker::WorkComplete() {
-  HandleScope scope(_env);
-  if (_error.IsEmpty()) {
-    OnOK();
-  }
-  else {
-    OnError(_error.Value());
-  }
-}
-
 inline ObjectReference& AsyncWorker::Persistent() {
   return _persistent;
 }
 
 inline void AsyncWorker::OnOK() {
-  _callback.MakeCallback(Env().Undefined(), std::vector<napi_value>());
+  _callback.MakeCallback(_persistent.Value(), {});
 }
 
 inline void AsyncWorker::OnError(Error e) {
-  _callback.MakeCallback(Env().Undefined(), std::vector<napi_value>({ e }));
+  _callback.MakeCallback(_persistent.Value(), { e });
 }
 
-inline void AsyncWorker::SetError(Error error) {
-  _error.Reset(error, 1);
+inline void AsyncWorker::SetError(std::string error) {
+  _error = error;
 }
 
 inline void AsyncWorker::OnExecute(napi_env env, void* this_pointer) {
   AsyncWorker* self = static_cast<AsyncWorker*>(this_pointer);
-  try {
-    self->Execute();
-  }
-  catch (const Error& e) {
-    self->SetError(e);
-  }
+  self->Execute();
```

Comment:
We could still store `error.what()` for any `std::exception` we run into … anything else is going to hard crash the process anyway, afaict?